hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -751,6 +752,15 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
       }
     }
 
+    // Check if the retry is with "Expect: 100-continue" header being present in the previous request.
```

Comment:
not sure about this strategy of patching/replacing the ua header

Reply:
"We are attempting to dynamically update the User-Agent header. The removal from the User-Agent would only be necessary if we are retrying the same append request that failed due to a user error. Consequently, the retry for the identical append request would occur without the 'Expect: 100-continue' header enabled." Hence the only way is to remove the string from the header. So even if we create a separate method that would have the same functionality. The other approach can be to add the string only if expectHeaderIsEnabled but since that is for each append request, it would need the userAgent header to be updated for each append request making the operation costly.

Reply:
> t would need the userAgent header to be updated for each append request making the operation costly.

is it expensive? as I've been wondering what it would take the patch the referrer header on every request so we can attach the same audit information we do for s3a calls and which are wonderful for support requests...

Reply:
If I were to update the User-Agent separately for each append call, it would require removing the existing User-Agent header and then updating it with a new value. This process can be resource-intensive. On the other hand, if I perform this modification only in one specific scenario, it would help minimize the overall cost.

Reply:
ok, though I'm surprised swapping the UA header is considered expensive.